### DISCUSSION TITLE
Trying to move windows using BetterTouchTools

### DISCUSSION THREAD

**Discussion by seishonagon** [ttscoff/bunch-docs#271] [Q&A]
I really like Moom, but BTT allows you to move windows from one desktop to another, whereas Moom ... won't 
So I've decided I wanted to give a try at using bunch with BTT to set up my apps the way I like. 

I've defined a set of keystrokes that allow BTT to move a window from one space to the next:
- Cmd-Ctl 3 will move the active window to space number 3
- Cmd-Ctl ➡️  will move the active window 1 space to the right 
- you get the picture 

I've also created "Named Triggers" in btt which you can call using a url scheme:
- open "btt://trigger_named/?trigger_name=move_window_4" will move the active window to space 4

I've tried doing these things in Bunch: 

```
# Move Nova window to space number 4
Nova^
open "btt://trigger_named/?trigger_name=move_desktop_4"
```

```
# Move Nova window to space number 4
Nova^
- {@^4}

```
```
# Move Nova window to space number 4
Nova^
{@^4}
```

None of them work. 
I'm assuming it's because I'm not directing the keystroke at the actual window, but at Nova itself, and it's trying to interpret a "command" rather than whatever magic BTT does to intercept the keystroke ...
And in the case of the named trigger, I imagine the window is not even remotely defined in the "open" command, so, no surprise there.

What could I do to mimick the live experience of actually pressing cmd, ctl, 4 at a window, regardless of the application it's running?

**Comment by ttscoff** [comment]
Ok, if you want to trigger the URL, you just put the bare url on a line, 
none of the `open ...` stuff needed. But that likely won't solve the 
issue.

The second attempt will send the keyboard shortcut to the app, which 
isn't registered to handle it.

The third attempt is the closest, where Nova would launch to foreground 
and then the key combo would be sent as a global shortcut. I could see 
that working occasionally but not consistently. You could possibly add 
this to a delayed snippet, having launched Nova previously in the Bunch 
to ensure it was already running.

```
Nova

<<#Focus Nova ~3
___
### [Focus Nova] ###
Nova^
{@^4}
```

Your shortcut depends on a specific window being key (focused) when it 
fires, which in the context of a Bunch is going to be tricky. You could 
put the trigger on a delay and launch sequentially, but it would be a 
bit hit or miss. I haven't used BTT for window management, is it 
possible to target a specific app and move windows? If so, you could 
make a Nova-specific shortcut that would work even if the window wasn't 
focused.

I would have to dig into BTT window management to offer a better answer, 
maybe someone else here has tips?

-Brett

**Comment by seishonagon** [comment]
Thanks for the detailed walkthrough. I don't think it's possible to target just the window in BTT, but I'll check on the forum over there.
